src/databricks/labs/ucx/aws/credentials.py

```diff
@@ -230,3 +230,30 @@ def run(self, prompts: Prompts, *, single_role=False, role_name="UC_ROLE", polic
 
         self._resource_permissions.create_uc_roles(iam_list)
         self._resource_permissions.save_uc_compatible_roles()
+
+    def delete_uc_roles(self, prompts: Prompts):
+        uc_roles = self._resource_permissions.load_uc_compatible_roles()
+        if len(uc_roles) == 0:
+            self._resource_permissions.save_uc_compatible_roles()
```

Comment:
Why would this happen? The UC roles should be saved already before deleting it or not?

Reply:
That is true. In case if the file was deleted and we reply on that file, we could re-generate the file and check again